pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -323,10 +326,83 @@ func (s *Service) UpdateMigration(ctx context.Context, id int64, cm cloudmigrati
 	return nil, nil
 }
 
-func (s *Service) GetMigrationDataJSON(ctx context.Context, id int64) ([]byte, error) {
+func (s *Service) RunMigration(ctx context.Context, id int64) (*cloudmigration.MigrateDataResponseDTO, error) {
+	logger := s.log.FromContext(ctx)
+
+	// Get migration to read the auth token
+	migration, err := s.GetMigration(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("migration get error: %w", err)
+	}
+	// get CMS path from the config
+	domain, err := s.parseCloudMigrationConfig()
+	if err != nil {
+		return nil, fmt.Errorf("config parse error: %w", err)
+	}
+	path := fmt.Sprintf("https://cms-%s.%s/cloud-migrations/api/v1/migrate-data", migration.ClusterSlug, domain)
+
+	// Get migration data JSON
+	body, err := s.getMigrationDataJSON(ctx)
+	if err != nil {
+		s.log.Error("error getting the json request body for migration run", "err", err.Error())
+		return nil, fmt.Errorf("migration data get error: %w", err)
+	}
+
+	// Send the request to cms with the associated auth token
+	req, err := http.NewRequest(http.MethodPost, path, bytes.NewReader(body))
```

Comment:
Would having a cms api client and using it instead of building the http request here make it easier to read?

Reply:
100%, done! [0635fac](https://github.com/grafana/grafana/pull/86406/commits/0635faceb4fb47dc72ae87ee27475214b55f40e1)